packages/smooth_app/lib/background/background_task_details.dart

```diff
@@ -113,12 +114,31 @@ class BackgroundTaskDetails extends AbstractBackgroundTask {
   /// Uploads the product changes.
   @override
   Future<void> upload() async {
-    // TODO(AshAman999): check returned Status
-    await OpenFoodAPIClient.saveProduct(
+    if (_product.packagings != null) {
+      // For the moment, we can only save "packagings" with V3,
+      // and V3 can only save "packagings".
+      final ProductResultV3 result =
+          await OpenFoodAPIClient.temporarySaveProductV3(
+        getUser(),
+        _product.barcode!,
+        packagings: _product.packagings,
+        language: getLanguage(),
+        country: getCountry(),
+      );
+      if (result.status != ProductResultV3.statusSuccess &&
+          result.status != ProductResultV3.statusWarning) {
+        throw Exception('Could not save product - ${result.errors}');
+      }
+      return;
+    }
+    final Status status = await OpenFoodAPIClient.saveProduct(
       getUser(),
       _product,
       language: getLanguage(),
       country: getCountry(),
     );
+    if (status.status != 1) {
```

Comment:
Hello @M123-dev 
Just  out of curiosity, Can / Should we log this into sentry, Just wanted to make sure and keep a look into how many fails we do incur when saving data, would be helpful in understanding the background sync bug

Reply:
@AshAman999 More broadly, the question is again: what should we do with failing changes? Run them forever? Ignore them? Alert the user?

Reply:
I'm thinking that a alert would be more subtle for now
At least the user will know that something went wrong 

About retires I am not sure that we are going to have so many of those occurrences for failures,

I guess next setry report will reveal these data